File: mobile/lib/models/question.dart
class Question {
  final int id;
  final String sessionId;
  final int sequenceNo;
  final String subject;
  final String questionType;
  final String questionText;
  final String studentAnswer;
  final String correctAnswer;
  final String? imageUrl;
  final List<String>? knowledgePoints;
  final CorrectionResult? correction;

  Question({
    required this.id,
    required this.sessionId,
    required this.sequenceNo,
    required this.subject,
    required this.questionType,
    required this.questionText,
    required this.studentAnswer,
    required this.correctAnswer,
    this.imageUrl,
    this.knowledgePoints,
    this.correction,
  });

  factory Question.fromJson(Map<String, dynamic> json) => Question(
        id: json['id'],
        sessionId: json['session_id'] ?? '',
        sequenceNo: json['sequence_no'] ?? 0,
        subject: json['subject'] ?? 'other',
        questionType: json['question_type'] ?? 'other',
        questionText: json['question_text'] ?? '',
        studentAnswer: json['student_answer'] ?? '',
        correctAnswer: json['correct_answer'] ?? '',
        imageUrl: json['image_url'],
        knowledgePoints: (json['knowledge_points'] as List?)?.cast<String>(),
        correction: json['correction'] != null
            ? CorrectionResult.fromJson(json['correction'])
            : null,
      );
}

class CorrectionResult {
  final int id;
  final int questionId;
  final String status;
  final bool isCorrect;
  final String standardAnswer;
  final String solutionSteps;
  final String explanation;
  final String? errorReason;
  final String errorAnalysis;
  final double? aiConfidence;

  CorrectionResult({
    required this.id,
    required this.questionId,
    required this.status,
    required this.isCorrect,
    required this.standardAnswer,
    required this.solutionSteps,
    required this.explanation,
    this.errorReason,
    required this.errorAnalysis,
    this.aiConfidence,
  });

  factory CorrectionResult.fromJson(Map<String, dynamic> json) =>
      CorrectionResult(
        id: json['id'],
        questionId: json['question_id'],
        status: json['status'] ?? 'pending',
        isCorrect: json['is_correct'] ?? false,
        standardAnswer: json['standard_answer'] ?? '',
        solutionSteps: json['solution_steps'] ?? '',
        explanation: json['explanation'] ?? '',
        errorReason: json['error_reason'],
        errorAnalysis: json['error_analysis'] ?? '',
        aiConfidence: json['ai_confidence']?.toDouble(),
      );
}

class QuestionRegion {
  final int questionId;
  final double left;
  final double top;
  final double width;
  final double height;

  QuestionRegion({
    required this.questionId,
    required this.left,
    required this.top,
    required this.width,
    required this.height,
  });

  Map<String, dynamic> toJson() => {
        'question_id': questionId,
        'bbox': {
          'left': left,
          'top': top,
          'width': width,
          'height': height,
        },
      };
}

File: server/app/schemas/question.py
from pydantic import BaseModel
from datetime import datetime


class QuestionCreate(BaseModel):
    session_id: str
    sequence_no: int = 0
    subject: str = "other"
    question_type: str = "other"
    image_url: str | None = None
    region_bbox: dict | None = None


class QuestionResponse(BaseModel):
    id: int
    session_id: str
    sequence_no: int
    subject: str
    question_type: str
    question_text: str
    student_answer: str
    correct_answer: str
    image_url: str | None
    knowledge_points: list | None
    created_at: datetime

    model_config = {"from_attributes": True}


class CorrectionResponse(BaseModel):
    id: int
    question_id: int
    status: str
    is_correct: bool
    standard_answer: str
    solution_steps: str
    explanation: str
    error_reason: str | None
    error_analysis: str
    knowledge_points: list | None
    ai_confidence: float | None
    created_at: datetime

    model_config = {"from_attributes": True}


class OCRRequest(BaseModel):
    image_base64: str
    detect_handwriting: bool = True


class OCRResponse(BaseModel):
    questions: list[dict]
    raw_text: str
    regions: list[dict]


class VisionCorrectRequest(BaseModel):
    image_base64: str
    question_text: str = ""


class VisionCorrectResponse(BaseModel):
    question_text: str
    student_answer: str
    status: str
    is_correct: bool
    score: int = 0
    standard_answer: str
    solution_steps: str
    explanation: str
    error_reason: str | None = None
    error_analysis: str = ""
    knowledge_points: list[str] = []
    ai_confidence: float | None = None


class FingerPointRequest(BaseModel):
    image_base64: str
    question_regions: list[dict]


class FingerPointResponse(BaseModel):
    detected: bool
    finger_tip_x: float | None = None
    finger_tip_y: float | None = None
    pointed_question_id: int | None = None
    pointed_region: dict | None = None
    confidence: float = 0.0


class AIExplanationRequest(BaseModel):
    question_id: int


class AIExplanationResponse(BaseModel):
    question_id: int
    explanation_text: str
    solution_steps: str
    knowledge_points: list[str]
    tips: list[str]


class SimilarQuestionRequest(BaseModel):
    question_id: int
    count: int = 3


class SimilarQuestionResponse(BaseModel):
    original_question_id: int
    similar_questions: list[dict]
